src/common/transformations/src/transformations/common_optimizations/matmul_split_decomposition.cpp

```diff
@@ -0,0 +1,206 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/matmul_split_decomposition.hpp"
+
+#include <cstdint>
+#include <limits>
+#include <memory>
+#include <openvino/core/rt_info.hpp>
+#include <openvino/opsets/opset13.hpp>
+#include <openvino/opsets/opset6.hpp>
+#include <openvino/opsets/opset8.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <transformations/utils/utils.hpp>
+#include <vector>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/core/validation_util.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/gather.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/util/gather_base.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/type_relaxed.hpp"
+#include "transformations/utils/utils.hpp"
+
+using namespace ov::op;
+using namespace ov;
+using namespace ov::pass::pattern;
+
+void pass::MatmulGatherDecomposition::split_weights(const Output<Node>& weights,
+                                                    OutputVector& new_weights,
+                                                    const bool& have_bias,
```

Comment:
```suggestion
                                                    const bool have_bias,
```

Reply:
Or probably we could remove this argument by making argument "bias" type of Node*, and check if it is nullptr.

Reply:
Done